File: app/src/main/java/com/socket/socket/utility/LoginUtility.java
package com.socket.socket.utility;

import androidx.annotation.NonNull;

import com.google.firebase.database.DataSnapshot;
import com.google.firebase.database.DatabaseError;
import com.google.firebase.database.ValueEventListener;
import com.google.firebase.database.annotations.NotNull;
import com.socket.socket.data.SharedPrefs;
import com.socket.socket.firebase.FirebaseClass;

import java.math.BigInteger;
import java.security.MessageDigest;
import java.security.NoSuchAlgorithmException;

public class LoginUtility{
    public static String getUsername(){
        return SharedPrefs.getUsername();
    }

    public static boolean isUser(DataSnapshot d){
        return d.hasChild("username");
    }

    public static boolean isLoggedIn(){
        return !SharedPrefs.getUsername().equals("null");
    }

    public static void updateEmail() {
        FirebaseClass.getDBRef().child(SharedPrefs.getEmail()).addListenerForSingleValueEvent(new ValueEventListener() {
            @Override
            public void onDataChange(DataSnapshot snapshot) {
                for(DataSnapshot d : snapshot.getChildren()) {
                    if(d.getKey().equals("email")){
                        SharedPrefs.setEmail(String.valueOf(d.getValue()));
                        break;
                    }
                }
            }

            @Override
            public void onCancelled(@NonNull @NotNull DatabaseError error) {

            }
        });
    }
}

File: app/src/main/java/com/socket/socket/utility/LoginUtility.kt
package com.socket.socket.utility

import com.google.firebase.database.DataSnapshot
import com.google.firebase.database.DatabaseError
import com.google.firebase.database.ValueEventListener
import com.google.firebase.database.annotations.NotNull
import com.socket.socket.data.SharedPrefs
import com.socket.socket.firebase.FirebaseClass

object LoginUtility {
    fun getUsername(): String? {
        return SharedPrefs.getUsername()
    }

    fun isUser(d: DataSnapshot?): Boolean {
        return d.hasChild("username")
    }

    fun isLoggedIn(): Boolean {
        return SharedPrefs.getUsername() != "null"
    }

    fun updateEmail() {
        FirebaseClass.getDBRef().child(SharedPrefs.getEmail())
            .addListenerForSingleValueEvent(object : ValueEventListener {
                override fun onDataChange(snapshot: DataSnapshot?) {
                    for (d in snapshot.getChildren()) {
                        if (d.key == "email") {
                            SharedPrefs.setEmail(d.value.toString())
                            break
                        }
                    }
                }

                override fun onCancelled(@NotNull error: DatabaseError) {}
            })
    }
}